From a185d0bdeba979df5359f4cdc46cc4b6e49669ca Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 11 Aug 2014 12:26:35 -0400
Subject: [mfd] lpc_ich: Fix ACPI enable bitmask

Message-id: <1407759999-16143-5-git-send-email-prarit@redhat.com>
Patchwork-id: 87598
O-Subject: [RHEL7.1 PATCH BZ 1084134 4/8] mfd: lpc_ich: Fix ACPI enable bitmask
Bugzilla: 1084134
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1084134

commit f5dccb15877b82a40950c6f752d5345c86189fc9
Author: Peter Tyser <ptyser@xes-inc.com>
Date:   Mon Mar 10 16:34:51 2014 -0500

    mfd: lpc_ich: Fix ACPI enable bitmask

    The original bitmask of 0x10 was incorrect and would result in a write
    to a reserved read-only bit instead of enabling the ACPI I/O
    region.  Update it to the proper value of 0x80.

    Signed-off-by: Peter Tyser <ptyser@xes-inc.com>
    Tested-by: Rajat Jain <rajatjain@juniper.net>
    Reviewed-by: Guenter Roeck <linux@roeck-us.net>
    Signed-off-by: Lee Jones <lee.jones@linaro.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/mfd/lpc_ich.c b/drivers/mfd/lpc_ich.c
index 4f59b8d..315a960 100644
--- a/drivers/mfd/lpc_ich.c
+++ b/drivers/mfd/lpc_ich.c
@@ -760,7 +760,7 @@ static void lpc_ich_enable_acpi_space(struct pci_dev *dev)
  u8 reg_save;
 
  pci_read_config_byte(dev, priv->acpi.ctrl, &reg_save);
- pci_write_config_byte(dev, priv->acpi.ctrl, reg_save | 0x10);
+ pci_write_config_byte(dev, priv->acpi.ctrl, reg_save | 0x80);
  priv->acpi.save = reg_save;
 }
 
-- 
1.7.1